src/gl-state-egl.cpp

```diff
@@ -428,6 +428,7 @@ GLStateEGL::gotValidDisplay()
     if (egl_display_)
         return true;
 
+#if defined(PFNEGLGETPLATFORMDISPLAYEXTPROC)
```

Comment:
PFNEGLGETPLATFORMDISPLAYEXTPROC is a typedef so the preprocessor check doesn't have the desired effect.

My suggestion would be to update gl-headers.h to provide our own version of the PFNEGLGETPLATFORMDISPLAYEXTPROC typedef there. We can guard our typedef with "#ifndef EGL_EXT_platform_base", although that's not strictly required since we shouldn't be changing the typedef type anyway.

This way we avoid the preprocessor check in gl-state-egl.cpp. I assume that the EGL platform this workaround is for does not advertise "EGL_EXT_platform_base" in the EGL extensions, so things should work properly (i.e. we fall back to eglGetDisplay) even with PFNEGLGETPLATFORMDISPLAYEXTPROC defined by us.

Reply:
I don't think gl-headers.h is the place, since eglext.h isn't included from there (so we can't make the decision that early).